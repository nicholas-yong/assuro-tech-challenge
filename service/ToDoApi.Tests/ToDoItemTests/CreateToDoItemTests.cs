using System;
using System.Threading.Tasks;
using HotChocolate.Execution;
using NUnit.Framework;
using Snapshooter.NUnit;

namespace ToDoApi.Tests.ToDoItemTests
{
    public class CreateToDoItemTests: BaseTest
    {
        [Test]
        public async Task CanCreateNewItem()
        {
            // act
            IExecutionResult result = await Services.ExecuteRequestAsync(
                QueryRequestBuilder.New()
                    .SetQuery(@"
                        mutation
                        {
                            createToDoItem(input: {
                                content: ""My fourth todo""
                            }) {
                                toDoItem {
                                    content
                                    createdDate
                                    id
                                    status
                                }
                            }
                        }")
                    .Create());
            
            // assert
            result.MatchSnapshot(match => match
                .IgnoreField<Guid>("Data.createToDoItem.toDoItem.id")
                .IgnoreField<DateTimeOffset>("Data.createToDoItem.toDoItem.createdDate")
            );
        }
    }
}
