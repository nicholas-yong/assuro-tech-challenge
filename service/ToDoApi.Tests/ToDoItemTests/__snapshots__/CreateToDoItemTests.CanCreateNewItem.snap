{
  "Label": null,
  "Path": null,
  "Data": {
    "createToDoItem": {
      "toDoItem": {
        "content": "My fourth todo",
        "createdDate": "2021-02-17T03:30:50.359Z",
        "id": "ee219fb0d5914c88b503df394cf9c9da",
        "status": "PENDING"
      }
    }
  },
  "Errors": null,
  "Extensions": null,
  "ContextData": null,
  "HasNext": null
}
